cookbooks/kinematically_driven_subduction_2d/doc/kinematically_driven_subduction_2d.md

```diff
@@ -1,102 +1,194 @@
+not coupled to the mechanical evolution. A high conductivity (k = 183.33 Wm-1K-1) 
+is defined for the upper mantle to enforce the mantle adiabat in the slow convection
+system.
+
+```{literalinclude} Case2a_materialmodel.prm
+
+```
+
+We can monitor the temperature evolution, however, by tracking the
+root-mean-square (RMS) temperature and the isotherm depth using 
+two plugins called "composition_trms_statistics" and "isotherm depth".
+To this end, we add the postprocessors ’temperature statistics’ 
+and "isotherm depth" to the prm file. The RMS temperature is tracked 
+over the subducting plate (comprising BOC_SP, SHB_SP, thermal_SP) 
+and the whole domain over time. And the isotherm temperature to be 
+tracked is prescribed as 800 C.
+
+```{literalinclude} Case2a_postprocessing.prm
+
+```
+
+{numref}`fig:QQ_case2a_diagnostics` shows the RMS temperature, 
+800 C isotherm depth as well as the other diagnostics. The 800 C isotherm 
+keeps going deeper, while in the end the descend speed becomes slower. 
+The RMS temperature of the whole domain keeps decreasing due to the 
+input of cold subducting plate into model domain. The slab RMS temperature 
+cools at the initial stage, then it goes up due to being warmed up by the 
+mantle. Notice that these other diagnostics are indeed the same as for Case 1.
+
+```{figure-md} fig:QQ_case2a_diagnostics
+<img src="Case2a_diagnostics.*" />
+
+ ASPECT Case 2a and case 1, Elefant case 2a diagnostic quantities
+```
+
+## Case 2b: Model with a temperature-dependent density
+
+Case2b builds on case 2a.  Compared to Case2a, the thermal expansivity in Case 2b
+is set to 2.5e-5 K<sup>-1</sup> instead of 0. Therefore, temperature feeds into density 
+and therefore stokes equations. In addition, the benchmark uses different reference temperatures for each composition, but only one value of reference temperature is 
+allowed to be given to various compositional fields in ASPECT. Therefore, we adapted reference densities for different compositional fields. All the other parameters are 
+identical with case2a:
+
+```{literalinclude} kinematically_driven_subduction_2d_case2b.prm
+
+```
+The density evolution can be seen in {numref}`fig:QQ_case2b_density_evolution`. When the slab warms up, the density contrast between the slab and the mantle decreases.
+
+```{figure-md} fig:QQ_case2b_density_evolution
+<img src="Case2b_density_evolution.*" />
+ Case2b density evolution
+```
+Therefore, the  subduction in this case is much slower than that in case1 and 2b (see {numref}`fig:QQ_case2b_diagnostics`). The subducting plate did not reach 
+the bottom till 15 Myr, i.e., the third stage in case 1 and 2b is missing in this 
+case. 
+
+```{figure-md} fig:QQ_case2b_diagnostics
+<img src="Case2b_diagnostics.*" />
+
+ ASPECT Case2b diagnostic quantities
+```
+
+The root mean square velocity over the whole domain increases (see {numref}`fig:QQ_case2b_diagnostics`) .This is due to higher velocities in the 

```

Comment:
```suggestion
The RMS velocity over the whole domain continuously increases (see {numref}`fig:QQ_case2b_diagnostics`). This is due to higher sub-lithospheric velocities on the 

```

Reply:
Do you mean to explain here why the RMS velocity is higher than in the previous cases? Or just that it keeps increasing because the third phase is not reached yet? Please clarify in the text.